### PR TITLE
on_success_script changes

### PR DESCRIPTION
**Pull request by robolson**
All uses of on_success_script so far have been to push additional branches based on the contents of the commit. `inside_repo` has access to the remote whereas `inside_copy` does not.

WARNING: This is a *breaking change* if you are currently using a on_success_script. Kochiku will no longer run on_success_script with the working directory checked out to the commit that was built. Instead, the script is run from inside a git checkout of the repo but it should be treated as bare repo. Avoid accessing files on the file system and instead rely solely on git commands.

### CODE REVIEW

File: spec/lib/build_strategies/production_build_strategy_spec.rb [hunk 1]
```diff
@@ -115,7 +115,6 @@
     }
 
     before do
-      allow(GitRepo).to receive(:inside_copy).and_yield
```
Comment (moshez) [inline spec/lib/build_strategies/production_build_strategy_spec.rb]: Why is this deleted rather than changing it to "inside_repo"?

Comment (robolson) [inline spec/lib/build_strategies/production_build_strategy_spec.rb]: `inside_repo` is already stubbed in the setup for the file. It's not obvious I know.

**Comment by moshez** [comment]
LGTM

**Comment by robolson** [comment]
The changes in this PR were reversed and improved upon in #200. The _breaking changes_ warning in this PR can be ignored.
